File: src/main/java/com/example/Toros/DTO/UserProfileDto.java
package com.example.Toros.DTO;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

/**
 * Data Transfer Object for User Profile.
 * This class is used to transfer user profile data between the client and server.
 */
@Data
public class UserProfileDto {

    @NotBlank(message = "First name must not be blank")
    private String firstName;

    @NotBlank(message = "Last name must not be blank")
    private String lastName;

    @NotBlank(message = "Email must not be blank")
    @Email(message = "Email should be valid")
    private String email;

    // optional field: allow blank
    private String phone;

    @Size(max = 500, message = "Bio must not exceed 500 characters")
    private String bio;

    @NotBlank(message = "Skill level must not be blank")
    private String skillLevel;

    // getters + setters omitted for brevity
}

File: system/backend/API/DTOs/UserProfileDto.cs
namespace Toros.Backend.DTOs;

public class UserProfileDto
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Bio { get; set; } = string.Empty;
    public string Picture { get; set; } = string.Empty;
    public string SkillLevel { get; set; } = "Beginner";
}
